### ISSUE TITLE
Assert failure when calling arbitrary executables with `os.execute`

### ISSUE THREAD

**Issue by atusy** [neovim/neovim#21671]
### Describe the bug

Evaluating Lua function `os.execute` with arbitrary executables, echo for example, terminates nvim, and nvim returns exit code 134.

Neovim also shows the following message.

```
nvim: /home/atusy/.asdf/downloads/neovim/ref-master/src/nvim/rbuffer.c:144: rbuffer_consumed: Assertion `count && count <= buf->size' failed.                                                                                                                                          zsh: IOT instruction (core dumped)  nvim -u NORC --clean +"lua os.execute[[echo]]"
```

### Steps to reproduce

``` bash
nvim -u NORC --clean +"lua os.execute[[echo]]"
```

### Expected behavior

`os.execute` should return the exit code without terminating nvim.

### Neovim version (nvim -v)

v0.9.0-dev commit f7ad46e69ed407579517694ccf3dc1beffe7acdc

### Vim (not Nvim) behaves the same?

no

### Operating system/version

Manjaro Linux

### Terminal name/version

wezterm 20220327-162807-53fc926b

### $TERM environment variable

xterm-256color

### Installation

asdf install neovim ref:master

**Comment by lewis6991** [comment]
This is a known issue. Don't use `os.execute`, just use `vim.fn.system` or `vim.loop.spawn`.

**Comment by atusy** [comment]
Thanks. I will follow your suggestion.

BTW, I've forgotten to mention that v0.8.2 works fine with `os.execute`.

**Comment by zeertzjq** [comment]
Fixed by #21675

**Comment by atusy** [comment]
@bfredl Perfect!! Thank you.
